mirdata/datasets/beatles.py

```diff
@@ -141,151 +138,117 @@ def to_jams(self):
+    reader = csv.reader(fbeats, dialect)
+    for line in reader:
+        beat_times.append(float(line[0]))
+        beat_positions.append(line[-1])
 
     beat_positions = _fix_newpoint(np.array(beat_positions))
     # After fixing New Point labels convert positions to int
-    beat_positions = [int(b) for b in beat_positions]
-
-    beat_data = annotations.BeatData(np.array(beat_times), np.array(beat_positions))
+    beat_data = annotations.BeatData(
+        np.array(beat_times), np.array([int(b) for b in beat_positions])
+    )
 
     return beat_data
 
 
-def load_chords(chords_path):
+@io.coerce_to_io(io.Mode.TEXT)
+def load_chords(fchords: TextIO) -> annotations.ChordData:
     """Load Beatles format chord data from a file
 
     Args:
-        chords_path (str): path to chord annotation file
+        fchords (str or file-like): path or file-like object pointing to a chord annotation file
 
     Returns:
         ChordData: loaded chord data
 
     """
-    if chords_path is None:
-        return None
-
-    if not os.path.exists(chords_path):
-        raise IOError("chords_path {} does not exist".format(chords_path))
-
     start_times, end_times, chords = [], [], []
-    with open(chords_path, "r") as f:
-        dialect = csv.Sniffer().sniff(f.read(1024))
-        f.seek(0)
-        reader = csv.reader(f, dialect)
-        for line in reader:
-            start_times.append(float(line[0]))
-            end_times.append(float(line[1]))
-            chords.append(line[2])
-
-    chord_data = annotations.ChordData(np.array([start_times, end_times]).T, chords)
+    dialect = csv.Sniffer().sniff(fchords.read(1024))
+    fchords.seek(0)
+    reader = csv.reader(fchords, dialect)
+    for line in reader:
+        start_times.append(float(line[0]))
+        end_times.append(float(line[1]))
+        chords.append(line[2])
 
-    return chord_data
+    return annotations.ChordData(np.array([start_times, end_times]).T, chords)
 
 
-def load_key(keys_path):
+@io.coerce_to_io(io.Mode.TEXT)
+def load_key(fkeys: TextIO) -> annotations.KeyData:
     """Load Beatles format key data from a file
 
     Args:
-        keys_path (str): path to key annotation file
+        fkeys (str or file-like): path or file-like object pointing to a key annotation file
 
     Returns:
         KeyData: loaded key data
 
     """
-    if keys_path is None:
-        return None
-
-    if not os.path.exists(keys_path):
-        raise IOError("keys_path {} does not exist".format(keys_path))
-
     start_times, end_times, keys = [], [], []
-    with open(keys_path, "r") as fhandle:
-        reader = csv.reader(fhandle, delimiter="\t")
-        for line in reader:
-            if line[2] == "Key":
-                start_times.append(float(line[0]))
-                end_times.append(float(line[1]))
-                keys.append(line[3])
-
-    key_data = annotations.KeyData(np.array([start_times, end_times]).T, keys)
+    reader = csv.reader(fkeys, delimiter="\t")
```

Comment:
what happens if `fkeys` is a string and not a file object? I don't think csv.reader can take strings directly

Reply:
oh is this what your decorator function is doing? I'm up for it, but I think it'll be yet another weird mirdata thing that it hard to pick up for new contributors - could this instead call a function that takes a string or file object and returns an object? what do you think is cleaner?

Reply:
We could change it to that, but then we have to deal with the whole "how do we handle None" business and that makes these functions a little messy again. The reason why I didn't remove the types from the doc string is that it better signifies what the user can pass to the function.

If these functions didn't have to return optional then switching to calling a function (which is what the decorator is kind of doing) would be the same number of lines

Reply:
My worry would be more for new users than new contributors with the decorator. New contributors should probably find this a bit easier no? With the decorator would make it so they never have to worry about file handling. They only have to worry about handling the contents of a file.

Reply:
True! I opened #398 to make this decorator and others clear for contributors.